### PR TITLE
feat: enable clippy::correctness as part of CI

### PR DESCRIPTION
**Pull request by pugachAG**
This enables clippy `clippy::correctness` checks as part of CI as discussed [here](https://github.com/near/nearcore/pull/8291#issuecomment-1372547523).

It is added as part of `sanity check` step for consistency, since that is where `cargo check` is executed.

This fixes the following issues in order to pass the added checks:
* noop `.clone()`
* this loop never actually loops 
* `0` is the minimum value for this type, the case where the two sides are not equal never occurs, consider using `self.burst == 0` instead

This PR is a part of #8145

### CODE REVIEW

File: chain/chunks/src/lib.rs [hunk 1]
```diff
@@ -612,10 +612,8 @@ impl ShardsManager {
                 continue;
             }
 
-            let fetch_from = if request_from_archival {
-                shard_representative_target.clone()
-            } else if we_own_part {
-                // If missing own part, request it from the chunk producer / node tracking shard
+            // If missing own part, request it from the chunk producer / node tracking shard
+            let fetch_from = if request_from_archival || we_own_part {
```
Comment (nagisa) [inline chain/chunks/src/lib.rs:616]: I think this change should be undone and `#[allowed]`. See https://github.com/rust-lang/rust-clippy/issues/3770 / https://github.com/rust-lang/rust-clippy/issues/5940 for some context too.

**Review by nagisa — COMMENTED** [review]
The clippy file has been added empty, so we should probably not add one at all.

File: chain/chain/src/store_validator/validate.rs [hunk 2]
```diff
@@ -890,7 +890,7 @@ pub(crate) fn block_height_cmp_tail_final(
 pub(crate) fn tx_refcount_final(sv: &mut StoreValidator) -> Result<(), StoreValidatorError> {
     let len = sv.inner.tx_refcount.len();
     if len > 0 {
-        for tx_refcount in sv.inner.tx_refcount.iter() {
+        if let Some(tx_refcount) = sv.inner.tx_refcount.iter().next() {
```
[the remark this answers is not in the document]
Comment (pugachAG) [reply chain/chain/src/store_validator/validate.rs]: yeah, `len` check is redundant, I've removed it in all instances

File: chain/chain/src/store_validator/validate.rs [hunk 3]
```diff
@@ -910,13 +910,13 @@ pub(crate) fn receipt_refcount_final(sv: &mut StoreValidator) -> Result<(), Stor
 pub(crate) fn block_refcount_final(sv: &mut StoreValidator) -> Result<(), StoreValidatorError> {
     if sv.inner.block_refcount.len() > 1 {
         let len = sv.inner.block_refcount.len();
```
[the remark this answers is not in the document]
Comment (pugachAG) [reply chain/chain/src/store_validator/validate.rs]: yep, done

File: chain/chain/src/store_validator/validate.rs [hunk 4]
```diff
@@ -890,7 +890,7 @@ pub(crate) fn block_height_cmp_tail_final(
 pub(crate) fn tx_refcount_final(sv: &mut StoreValidator) -> Result<(), StoreValidatorError> {
     let len = sv.inner.tx_refcount.len();
     if len > 0 {
-        for tx_refcount in sv.inner.tx_refcount.iter() {
+        if let Some(tx_refcount) = sv.inner.tx_refcount.iter().next() {
             err!("Found {:?} Txs that are not counted, i.e. {:?}", len, tx_refcount);
```
[the remark this answers is not in the document]
Comment (pugachAG) [reply chain/chain/src/store_validator/validate.rs]: done

File: chain/chunks/src/lib.rs (hunk 1 above)
Comment (pugachAG) [reply chain/chunks/src/lib.rs]: added `allow` and a comment referencing the issue

File: .buildkite/pipeline.yml [hunk 5]
```diff
@@ -90,6 +90,9 @@ steps:
 
       ./scripts/formatting --check
 
+      # clippy adoption is in progress, see https://github.com/near/nearcore/issues/8145
+      cargo clippy -- -A clippy::all -D clippy::correctness
```
[the remark this answers is not in the document]
Comment (pugachAG) [reply .buildkite/pipeline.yml]: Ah, good point, I definitely missed that script.

I've updated the script as well as CI pipeline definition

**Review by nagisa — APPROVED** [review]

**Pull request merged**